### PR TITLE
feat: enforce pnpm 7.5.0

### PR DESCRIPTION
**Pull request by venarius**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This fixes some issues with pnpm that I had while creating PRs for Novu:
  ### 1. Newer versions of pnpm will break the CI and seem to install unneeded packages.
    Reference: https://github.com/novuhq/novu/pull/1122
  
    Solution: Enforce pnpm version of 7.5.0
    Screenshot 2022-08-25 144807
 
  &nbsp;
  ### 2. When running ```npm run setup:project``` for the first time everything works as inteded. But if you run that command again it would downgrade your pnpm-lock.yaml to lockfileVersion 5.3
  Reference inside https://github.com/novuhq/novu/pull/1048:
  
    Screenshot 2022-08-25 145941
    
    That was an issue because of the repos ```package.json``` having ```pnpm: 6.34.0``` as devDependency and npx falling back to the local cached version instead of using the version number provided in the cli command:
    Screenshot 2022-08-25 145211
    
    Solution: Increase the devDependency version of pnpm to 7.5.0

### CODE REVIEW

**Comment by venarius** [comment]
Reason for failing NX build command: https://github.com/nrwl/nx/issues/10111
Due to upgrade of pnpm from ```6.34.0``` to ```7.5.0``` as devDependency. Issue fixed with NX version ```14.5.10```

File: package.json [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
   "private": true,
+  "packageManager": "pnpm@7.5.0",
   "scripts": {
     "bootstrap": "npm run setup:dev",
     "start": "npm run jarvis",
     "preinstall": "npx only-allow pnpm",
     "prepare": "husky install",
     "publish": "lerna publish from-package",
-    "setup:project": "npx pnpm@7.5.0 i && node scripts/setup-env-files.js && npx pnpm build",
+    "setup:project": "pnpm i && node scripts/setup-env-files.js && pnpm build",
```
Comment (scopsy) [inline package.json:11]:
```suggestion
    "setup:project": "npx pnpm i && node scripts/setup-env-files.js && npm run build",
```

Wanted to have the initial setup of the project work without pnpm installed on your machine, and you would need it only if you want to add some dependencies. Wdyt, of only leaving the npx here?

File: package.json [hunk 2]
```diff
@@ -111,7 +112,7 @@
     "listr": "^0.14.3",
     "meow": "^10.1.1",
     "mississippi": "^4.0.0",
-    "pnpm": "^6.34.0",
+    "pnpm": "7.5.0",
```
Comment (scopsy) [inline package.json:115]: This is great!

File: package.json [hunk 3]
```diff
@@ -1,13 +1,14 @@
 {
   "name": "novuhq",
   "private": true,
+  "packageManager": "pnpm@7.5.0",
```
Comment (scopsy) [inline package.json:4]: Oh nice didn't knew this prop existed

**Review by scopsy — COMMENTED** [review]
This is great, a small comment on the setup:project command wdyt on that one?

File: package.json (hunk 1 above)
Comment (venarius) [reply package.json:11]: Yeah we can do that, true 🙂

Comment (scopsy) [reply package.json]: good point about the lock part. Agree with this.

**Comment by scopsy** [comment]
Oh Looks like vercel is failing :|

```




Cloning github.com/novuhq/novu (Branch: feat/enforce-pnpm-version, Commit: 310dec1)
23:36:18.232 | Cloning completed: 738.862ms
23:36:19.463 | Looking up build cache...
23:36:23.267 | Build cache downloaded [53.70 MB]: 3123.984ms
23:36:23.489 | Running "vercel build"
23:36:23.984 | Vercel CLI 28.1.2
23:36:24.537 | Detected `pnpm-lock.yaml` generated by pnpm 7...
23:36:24.537 | Running "install" command: `cd ../ && npm run install:docs`...
23:36:24.753 |  
23:36:24.754 | > novuhq@ install:docs /vercel/path0
23:36:24.754 | > pnpm install --filter @novu/docs
23:36:24.754 |  
23:36:25.319 | ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)
23:36:25.319 |  
23:36:25.320 | Your pnpm version is incompatible with "/vercel/path0".
23:36:25.320
```

We are looking right now to switch to netlify and they allow easily override the install script on their dashboard.

**Comment by scopsy** [comment]
@andrewgolovanov this one is another reason we want to make the move to netlify for all our projects.

**Comment by venarius** [comment]
@scopsy maybe you can use this? https://vercel.com/changelog/corepack-experimental-is-now-available

**Comment by scopsy** [comment]
It worked!

**Review by scopsy — APPROVED** [review]

**Pull request merged**